File: Controllers/OrderController.vb
Imports System.Web.Mvc
Imports My.Models

Namespace My.Controllers
    <Authorize>
    Public Class OrderController
        Inherits Controller

        ' GET: /Order
        Function Index(status As String, page As Integer?) As ActionResult
            Dim memberId = CInt(Session("MemberId"))
            Dim vm = New OrderListViewModel With {
                .PageIndex = If(page, 1),
                .PageSize = 10,
                .StatusFilter = status
            }
            ' TODO: 查詢該會員的訂單列表
            Return View(vm)
        End Function

        ' GET: /Order/Detail/5
        Function Detail(id As Integer) As ActionResult
            Dim memberId = CInt(Session("MemberId"))
            ' TODO: 查詢訂單詳細資料（需驗證為本人訂單）
            Dim order As Order = Nothing
            If order Is Nothing Then Return HttpNotFound()
            Return View(order)
        End Function

        ' POST: /Order/Cancel/5
        <HttpPost>
        <ValidateAntiForgeryToken>
        Function Cancel(id As Integer) As ActionResult
            ' TODO: 取消訂單（僅 Pending 狀態可取消）
            Return RedirectToAction("Detail", New With {.id = id})
        End Function

    End Class
End Namespace

File: Controllers/OrderController.cs
using System.Web.Mvc;
using MyWeb.Models;

namespace MyWeb.Controllers
{
    [Authorize]
    public class OrderController : Controller
    {
        // GET: /Order
        public ActionResult Index(string status, int? page)
        {
            var memberId = (int)Session["MemberId"];
            var vm = new OrderListViewModel
            {
                PageIndex = page ?? 1,
                PageSize = 10,
                StatusFilter = status
            };
            // TODO: 查詢該會員的訂單列表
            return View(vm);
        }

        // GET: /Order/Detail/5
        public ActionResult Detail(int id)
        {
            var memberId = (int)Session["MemberId"];
            // TODO: 查詢訂單詳細資料（需驗證為本人訂單）
            Order order = null;
            if (order == null) return HttpNotFound();
            return View(order);
        }

        // POST: /Order/Cancel/5
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult Cancel(int id)
        {
            // TODO: 取消訂單（僅 Pending 狀態可取消）
            return RedirectToAction("Detail", new { id = id });
        }
    }
}
